marklogic-data-hub/src/test/java/com/marklogic/hub/flow/AddSourceNameAndTypeStepDefTest.java

```diff
@@ -0,0 +1,124 @@
+package com.marklogic.hub.flow;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.marklogic.hub.AbstractHubCoreTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class AddSourceNameAndTypeStepDefTest extends AbstractHubCoreTest {
```

Comment:
These are really good tests, but I think we need a couple more things - need to verify that if "sources" already exists, then a new source is added and the existing ones aren't modified. You can simply add one more JSON doc and one more XML doc to your test data and verify them as part of the first test scenario (I don't think you need to verify them in every other test).

Reply:
I have run the same flow with different options on the same files without adding new files and verified if the previously added sources are not updated.